server/serverAPI/controller/AuthenticationController.ts

```diff
@@ -101,6 +118,93 @@ export default class AuthenticationController extends BaseUserController {
         }
     }
 
+    confirmVerificationCode = async (req: Request, res: Response) => {
+        let username = req.body?.username;
+        let inputCode = req.body?.verificationCode;
+
+        let actualCode = AuthenticationController.verificationCodesMap.get(username);
+
+        try {
+            await this.requestGet(new Map([["username", username]]), res)
+        } catch (response) {
+            return response;
+        }
+
+        if (actualCode === undefined) {
+            return this.sendError(400, res, "Verification code is either expired or not issued.");
+        }
+
+        if (inputCode !== actualCode) {
+            return this.sendError(400, res, "Verification code is invalid.");
+        }
+
+        try {
+            let user = await this.requestGet(new Map([["username", username]]), res);
```

Comment:
Instead of doing this lookup twice, once here and once above, can you store the user above if the request succeeds?